### DISCUSSION TITLE
Airtable Update query failing

### DISCUSSION THREAD

**Discussion by ghost** [Budibase/budibase#7170] [Help] [answered]
Hello 

I'm trying to set-up an airtable **update** query. I've followed the instructions in [the budibase docs](https://docs.budibase.com/docs/airtable) but I've had no luck in getting it to run. I've set-up the query based on the structure of my airtable api docs, but I keep getting an error related to the record ID. 

<img width="1570" alt="Screenshot 2022-08-09 at 11 02 16" src="https://user-images.githubusercontent.com/64409787/183621917-825f327c-bf14-4495-9884-bcf4ac19a976.png">

I've managed to get a **Read** query to work successfully, so I know my credentials etc are working. Have I set up the binding incorrectly?

Thanks!

**Comment by R2bEEaton** [comment]
(Not a solution)

I think the problem is with the Fields section - it's not a field for bindings as far as I'm aware. As per [the docs you linked](https://docs.budibase.com/docs/airtable#function-update) it seems you need to pass the Airtable record ID that you want to update in the `id` Field. When testing the query in the Data screen, you can simply input one manually, but on a page it doesn't seem like we have access to it.
![image](https://user-images.githubusercontent.com/34921506/183627792-46b664cb-cb4b-4721-8060-83e9b5b8be49.png)

Looks like it could be a bug, but I'll wait for other input.

**Reply by melohagan** [reply]
Hey @adamjdavies could you also manually replace the `{{id}}` in the Id field with the record id value and see if that's the same outcome, so that we can rule out a binding issue please?

**Reply by ghost** [reply]
Hi @melohagan 

Still getting the same result:
<img width="1569" alt="Screenshot 2022-08-09 at 12 02 41" src="https://user-images.githubusercontent.com/64409787/183632476-49924566-4fde-4e07-a5e0-a41378ae81cf.png">

I've tried all sorts of formats, including placing the full request body in the 'Query' section. Nothing seems to work. There does seem to be something happening in the background however, as if I use an incorrect record ID (in the screenshot below I've use a record from another table in the base), I'm getting this error:
<img width="1569" alt="Screenshot 2022-08-09 at 12 04 57" src="https://user-images.githubusercontent.com/64409787/183632768-dc72e10f-c519-4ea5-b33d-f80a9ee1e96e.png">

Any other info I can provide? I'm currently using version 1.2.23 and the link to this app is https://tern.budibase.app/app/tern

Thanks

**Comment by melohagan** [comment, marked as answer]
Thanks for the additional info @adamjdavies 

The issue is that you need to enter the table name, but there is no field to allow you to do that. I'm going to work on a fix for this now:
https://github.com/Budibase/budibase/issues/7177

**Reply by ghost** [reply]
Amazing, thank you for looking into this @melohagan

**Reply by ghost** [reply]
Hey @melohagan 

Any update on when this should be set live?

Thanks!

**Reply by melohagan** [reply]
This fix should be live now! **v1.2.36**

**Reply by ghost** [reply]
Thanks for the quick turn around!

I've just tested but I'm still getting the same error. I've tried both adding the recordD as a binding and adding it directly but neither works:
<img width="1549" alt="Screenshot 2022-08-11 at 10 22 57" src="https://user-images.githubusercontent.com/64409787/184103130-7acc06b6-30d8-4f38-b743-0a33efb7a107.png">

Have I missed something here?

Thanks

**Reply by melohagan** [reply]
It's expecting table name, not Id

**Reply by ghost** [reply]
Like this? Still getting the same record ID error no matter what I enter into the Table and Record ID fields.

<img width="1549" alt="Screenshot 2022-08-11 at 10 53 40" src="https://user-images.githubusercontent.com/64409787/184108582-32f65f5e-1d2b-41b4-be7d-89f55df522d5.png">

**Reply by melohagan** [reply]
Lets try to narrow this down then. Create a new **update** query. Don't add any bindings, just include your record id, table name, and a payload that would change a single property, e.g.
```json
{
  "Product title": "NEW_TITLE"
}
```

**Reply by ghost** [reply]
Ah, it's working now! Must be something wrong with my query. Will go through line by line to see where the problem lies. 

Thanks for the assistance!

**Reply by melohagan** [reply]
You're very welcome! Also sometimes when there is a bug fix, the query/datasource has to be recreated in order to pull in the change - although I don't think that should be the case in this instance.
